lib/ecto/query/builder/from.ex

```diff
@@ -62,6 +62,9 @@ defmodule Ecto.Query.Builder.From do
           # When a binary is used, there is no model
           {1, query(source, nil)}
 
+        {source, model} when is_binary(source) and is_atom(model) ->
```

Comment:
This code here is just an optimization. If it can't expand it at compile time, it will fallback to expand at runtime.